src/Moryx.Products.Management/Implementation/Storage/ProductStorage.cs

```diff
@@ -83,11 +95,15 @@ internal class ProductStorage : IProductSearchStorage
         /// </summary>
         public void Start()
         {
+            _productTypes = new List<Type>();
+            _recipeTypes = new List<Type>();
+
             // Create type strategies
             foreach (var config in Config.TypeStrategies)
             {
                 var strategy = StrategyFactory.CreateTypeStrategy(config);
                 TypeStrategies[config.TargetType] = strategy;
+                _productTypes.Add(strategy.TargetType);
```

Comment:
Maybe just `TypeStragies.Values.Select().ToArray` ?

Reply:
Is also possible, this way we save the additional iteration over all producttypestrategies and recipetypestrategies 😅

Reply:
True, but how often do we iterate and how many entries will this have

Reply:
It won't have a significant effect on the startup time, nevertheless I wouldn't have said that there is a more significant benefit in use the linq expression 😅 If you like it better I can certainly change it 😊

Reply:
I am not worried about start-up time. :-) Keeping two lists with similar information always comes with the risk of inconsistency. E.g. TypeStrategies gets modified and someone forget _productTypes.

Reply:
Ahh, yeah good point. This is an argument I was looking for 😄 I will change it 👍

Reply:
Otherwise I would keep it like a professor I admired for saying: If I can't tell you why it shouldn't be done that way, you should be allowed to do it your way